src/alchemlyb/parsing/namd.py

```diff
@@ -3,20 +3,103 @@
 """
 import pandas as pd
 import numpy as np
+import logging
 from .util import anyopen
 from . import _init_attrs
 from ..postprocessors.units import R_kJmol, kJ2kcal
 
+logger = logging.getLogger("alchemlyb.parsers.NAMD")
+
 k_b = R_kJmol * kJ2kcal
 
+
+def _get_lambdas(fep_files):
+    """Retrieves all lambda values included in the FEP files provided.
+    
+    We have to do this in order to tolerate truncated and restarted fepout files.
+    The IDWS lambda is not present at the termination of the window, presumably
+    for backwards compatibility with ParseFEP and probably other things.
+
+    For a given lambda1, there can be only one lambda_idws.
+
+    Parameters
+    ----------
+    fep_files: str or list of str
+        Path(s) to fepout files to extract data from.
+
+    Returns
+    -------
+    List of floats, or None if there is more than one lambda_idws for each lambda1.
+    """
+
+    lambda_fwd_map, lambda_bwd_map = {}, {}
+    is_ascending = set()
+    endpoint_windows = []
+
+    for fep_file in sorted(fep_files):
+        with anyopen(fep_file, 'r') as f:
+            for line in f:
+                l = line.strip().split()
+                if l[0] not in ['#NEW', '#Free']:
+                    continue
+
+                # We might not have a #NEW line so make the best guess
+                if l[0] == '#NEW':
+                    lambda1, lambda2 = float(l[6]), float(l[8])
+                    lambda_idws = float(l[10]) if 'LAMBDA_IDWS' in l else None
+                elif l[0] == '#Free':
```

Comment:
There is no `else` (and the case of neither `if` nor `elif` is never tested). Under which circumstances would `else` happen?

Reply:
This function only extracts the lambda values from a set of .fepout files. The lambda values are only in a small subset of lines in each file, so the `else` case is a `FepEnergy` or `FepE_back` line, which also contains an energy value for the timestep, which we don't care about in this function.

Reply:
Can you please also add an explanatory comment? That would be sufficient. Thanks.